java-squid/src/main/java/org/sonar/java/se/SymbolicValueRelation.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.se;
+
+import javax.annotation.Nullable;
+
+import java.util.Collection;
+
+public abstract class SymbolicValueRelation {
+
+  public static EqualRelation relationEqual(SymbolicValue v1, SymbolicValue v2) {
+    return v1.compareTo(v2) > 0 ? new EqualRelation(v2, v1) : new EqualRelation(v1, v2);
+  }
+
+  public static NotEqualRelation relationNotEqual(SymbolicValue v1, SymbolicValue v2) {
+    return v1.compareTo(v2) > 0 ? new NotEqualRelation(v2, v1) : new NotEqualRelation(v1, v2);
+  }
+
+  @Nullable
+  public static Boolean isFulfilled(SymbolicValueRelation relationToCheck, Collection<SymbolicValueRelation> knownRelations) {
+    for (SymbolicValueRelation otherRelation : knownRelations) {
+      Boolean validity = areConsistent(relationToCheck, otherRelation, knownRelations);
+      if (validity != null) {
+        return validity;
+      }
+    }
+    return null;
+  }
+
+  @Nullable
+  private static Boolean areConsistent(SymbolicValueRelation relationToCheck, SymbolicValueRelation knownRelation, Collection<SymbolicValueRelation> knownRelations) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "areConsistent" is 15 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)